cmd/promtool/main.go

```diff
@@ -496,6 +496,12 @@ func QueryInstant(url *url.URL, query, evalTime string, p printer) int {
 	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
 	val, _, err := api.Query(ctx, query, eTime) // Ignoring warnings for now.
 	cancel()
+
+	var apiErr *v1.Error
+	if errors.As(err, &apiErr) {
+		fmt.Fprintf(os.Stderr, "query error: %s: detail: %s", apiErr.Msg, apiErr.Detail)
```

Comment:
`v1.Error` actually has a `String()` method, which also includes the error `Type` field.

How about this one:
```Go
fmt.Fprintf(os.Stderr, "query error: %v (detail: %s)", apiErr, apiErr.Detail)
```

(Note the parentheses to reduce the number of colons a bit and also clearly separate the detail from the normal error string.)

Reply:
Should we have that in client_golang?

Reply:
I assume that you don't need the details usually, and that's why the `String` method is only putting the `Type` and `Msg` fields into the string representation.